spec/features/admin_claim_tasks_update_with_dqt_api_spec.rb

```diff
@@ -568,7 +568,7 @@ def task_outcome
         let(:data) do
           {
             qts_award_date: Date.new(
-              MathsAndPhysics.first_eligible_qts_award_year(claim.academic_year).start_year - 1.year,
+              MathsAndPhysics.first_eligible_qts_award_year(claim.academic_year).start_year - 1.year.to_i,
```

Comment:
This doesn't seem right.

This was working before but something about the Ruby upgrade created an error?

Seems like `.start_year` returns the wrong thing or has a method missing rather than year should be converted to an integer.

Reply:
@nomasprime here is an example of the error we get when upgrading from 2.6.7 to 2.7.0.

```ruby
10) Admin claim tasks update with DQT API with submitted claim with eligible qualifications except QTS award date admin claim tasks qualifications view doesn't show task outcome
Failure/Error:
qts_award_date: Date.new(
MathsAndPhysics.first_eligible_qts_award_year(claim.academic_year).start_year - 1.year,
9,
1
),

TypeError:
invalid year (not numeric)
# ./spec/features/admin_claim_tasks_update_with_dqt_api_spec.rb:513:in `initialize'
# ./spec/features/admin_claim_tasks_update_with_dqt_api_spec.rb:513:in `new'
# ./spec/features/admin_claim_tasks_update_with_dqt_api_spec.rb:513:in `block (5 levels) in <top (required)>'
# ./spec/features/admin_claim_tasks_update_with_dqt_api_spec.rb:69:in `block (2 levels) in <top (required)>'
# ./spec/support/bullet.rb:6:in `block (2 levels) in <top (required)>'
```

When I converted the `1.year` to an integer the specs (20 failures) are fixed.

What are you thoughts given the above?

Reply:
is this spec originally correct? as far as i can tell the date generated before the update would have been:

`Mon, 01 Sep -31554937`

Just running in console:

```
Date.new(
irb(main):022:1* MathsAndPhysics.first_eligible_qts_award_year.start_year - 1.year,
irb(main):023:1* 9,
irb(main):024:1* 1
irb(main):025:1> )
2021-08-23 14:43:42.174196 D [17145:70256177651720 policy_configuration.rb:22] (0.324ms) ActiveRecord -- PolicyConfiguration Load -- {:sql=>"SELECT \"policy_configurations\".* FROM \"policy_configurations\" WHERE \"policy_configurations\".\"policy_type\" = $1 LIMIT $2", :binds=>{:policy_type=>"MathsAndPhysics", :limit=>1}, :allocations=>11, :cached=>nil}
=> Mon, 01 Sep -31554937
```

The `first_year` generated in the spec and as above are both integer 2015

Reply:
@nomasprime This is how I tracked it down from a console session:

1.
```ruby
MathsAndPhysics.first_eligible_qts_award_year.start_year
2021-08-22 12:49:23.660308 D [16155:7840 policy_configuration.rb:22] (0.250ms) ActiveRecord -- PolicyConfiguration Load -- {:sql=>"SELECT \"policy_configurations\".* FROM \"policy_configurations\" WHERE \"policy_configurations\".\"policy_type\" = $1 LIMIT $2", :binds=>{:policy_type=>"MathsAndPhysics", :limit=>1}, :allocations=>9, :cached=>nil}
=> 2015
```

2. 
```ruby
MathsAndPhysics.first_eligible_qts_award_year.start_year - 1.year
2021-08-22 12:49:33.234909 D [16155:7840 policy_configuration.rb:22] (0.337ms) ActiveRecord -- PolicyConfiguration Load -- {:sql=>"SELECT \"policy_configurations\".* FROM \"policy_configurations\" WHERE \"policy_configurations\".\"policy_type\" = $1 LIMIT $2", :binds=>{:policy_type=>"MathsAndPhysics", :limit=>1}, :allocations=>8, :cached=>nil}
=> -1 years and 2015 seconds
```

3.
```ruby
(MathsAndPhysics.first_eligible_qts_award_year.start_year - 1.year).to_i
2021-08-22 12:50:13.139824 D [16155:7840 policy_configuration.rb:22] (0.359ms) ActiveRecord -- PolicyConfiguration Load -- {:sql=>"SELECT \"policy_configurations\".* FROM \"policy_configurations\" WHERE \"policy_configurations\".\"policy_type\" = $1 LIMIT $2", :binds=>{:policy_type=>"MathsAndPhysics", :limit=>1}, :allocations=>7, :cached=>nil}
=> -31554937
```

Thereby using `.to_i` converted the 1.year back to an integer

Reply:
@nomasprime @d-a-v-e Testing this today reveals the following:

specs spec/features/admin_claim_tasks_update_with_dqt_api_spec.rb:654
Run options: include {:locations=>{"./spec/features/admin_claim_tasks_update_with_dqt_api_spec.rb"=>[654]}}

Randomized with seed 42418

Admin claim tasks update with DQT API
with EarlyCareerPayments policy
with submitted claim
with eligible qualifications
except QTS award date
admin claim tasks view
claim.eligibility.first_eligible_itt_academic_year.start_year: 2018
claim.eligibility.first_eligible_itt_academic_year.start_year - 1: 2017
claim.eligibility.first_eligible_itt_academic_year.start_year - 1.year: -31554934
1.year: 31556952
1.year.class: ActiveSupport::Duration
claim.eligibility.first_eligible_itt_academic_year.start_year.class: Integer
shows qualifications no match

Finished in 0.73228 seconds (files took 2.29 seconds to load)
1 example, 0 failures

Randomized with seed 42418